src/static/helpers/slas.ts

```diff
@@ -0,0 +1,277 @@
+ * @param usid? - optional Unique Shopper Identifier to enable personalization
+ * @returns TokenResponse
+ */
+export async function loginGuestUser(
+  slasClient: ShopperLogin,
+  credentials: {
+    clientSecret: string;
+  },
+  usid?: string
+): Promise<ShopperLogin.TokenResponse> {
+  const authorization = `Basic ${stringToBase64(
+    `${slasClient.clientConfig.parameters.clientId}:${credentials.clientSecret}`
+  )}`;
+
+  const options = {
+    headers: {
+      Authorization: authorization,
+    },
+    body: {
+      grant_type: "client_credentials",
+      ...(usid && { usid: usid }),
+    },
+  };
+
+  return slasClient.getAccessToken(options);
+}
+
+/**
+ * A single function to execute the ShopperLogin Public Client Registered User B2C Login with proof key for code exchange flow as described in the [API documentation](https://developer.salesforce.com/docs/commerce/commerce-api/references?meta=shopper-login:Summary).
+ * @param slasClient - a configured instance of the ShopperLogin SDK client.
+ * @param credentials - the shopper username and password for login and client secret for additional authentication
+ * @param credentials.username - the id of the user to login with
+ * @param credentials.password - the password of the user to login with
+ * @param credentials.clientSecret - secret associated with client ID
+ * @param parameters - parameters to pass in the API calls.
+ * @param parameters.redirectURI - Per OAuth standard, a valid app route. Must be listed in your SLAS configuration. On server, this will not be actually called
+ * @param parameters.usid? - optional Unique Shopper Identifier to enable personalization
+ * @returns TokenResponse
+ */
+export async function loginRegisteredUserB2C(
+  slasClient: ShopperLogin,
+  credentials: {
+    username: string;
+    password: string;
+    clientSecret: string;
+  },
+  parameters: {
+    redirectURI: string;
+    usid?: string;
+  }
+): Promise<ShopperLogin.TokenResponse> {
+  const codeVerifier = createCodeVerifier();
+  const codeChallenge = await generateCodeChallenge(codeVerifier);
+
+  const authHeaderUserPass = `Basic ${stringToBase64(
+    `${credentials.username}:${credentials.password}`
+  )}`;
+
+  const optionsLogin = {
+    headers: {
+      Authorization: authHeaderUserPass,
+    },
+    body: {
+      code_challenge: codeChallenge,
+      channel_id: slasClient.clientConfig.parameters.siteId,
+      client_id: slasClient.clientConfig.parameters.clientId,
+      redirect_uri: parameters.redirectURI,
+      ...(parameters.usid && { usid: parameters.usid }),
+    },
+  };
+
+  // Create a copy in order to modify behavior of the fetch call
+  // We do not want to redirect so manually control redirect behavior
+  const slasClientCopy = new ShopperLogin(slasClient.clientConfig);
+  slasClientCopy.clientConfig.fetchOptions = {
+    ...slasClient.clientConfig.fetchOptions,
+    redirect: "manual",
+  };
+
+  const response = await slasClientCopy.authenticateCustomer(
```

Comment:
As above, there are many reasons why this call may fail. We'll want to inspect `response` for errors (bad status codes / lack of a redirect URL) and throw with enough information for the caller to resolve.